src/base/adaptive_playback.js

```diff
@@ -0,0 +1,102 @@
+import Playback from './playback'
+
+/**
+ * @typedef {Function} AdaptiveMediaActivatorFunction
+ * @function
+ * @param {Boolean} scheduleActivity Enable/disable activity, switch to a different media representation
+ * @param {Boolean} immediateFlush (default = false) Immediate switching, flushes playout buffer
+ * @returns {Boolean} Wether the switch request could be processed
+ *
+ */
+
+/**
+ * Video quality level description.
+ * In a set of quality levels, there should be exactly one of these objects representing each
+ * available quality at a given moment.
+ * @typedef {Object} VideoQualityLevel
+ * @class
+ * @property {String} id
+ * @property {Boolean} active
+ * @property {String} language
+ * @property {Number} width pixel
+ * @property {Number} height pixels
+ * @property {Number} bitrate bits/s
+ * @property {String} codec
+ * @member {AdaptiveMediaActivatorFunction} setActive
+ *
+ */
+
+/**
+ * Audio option available.
+ * @typedef {Object} AudioOption
+ * @class
+ * @property {String} id
+ * @property {Boolean} active
+ * @property {Number} volume
+ * @property {String} language
+ * @property {String} codec
+ * @property {Number} channels
+ * @property {String[]} roles
+ * @member {AdaptiveMediaActivatorFunction} setActive
+ */
+
+/**
+ * Closed caption option available.
+ * @typedef ClosedCaptionOption
+ * @class
+ * @property {String} id
+ * @property {Boolean} active
+ * @property {String} language
+ * @property {String[]} roles
+ * @member {AdaptiveMediaActivatorFunction} setActive
+ */
+
+export default class AdaptivePlayback extends Playback {
```

Comment:
I'm not sure if 'AdaptivePlayback' is the best name given this contains audio options and closed captions?

Reply:
I think these could be called MediaOptions. That's something like we're doing in the iOS/Android native versions of Clappr.

Reply:
@tjenkinson Adaptive is not only about (video) quality, but generally having content that adapts to the viewers needs/choices.

Reply:
@towerz To what did you refer exactly when stating the `MediaOptions` in iOS/Android Clappr? :)

Reply:
@tchakabam actually I was talking about audio/subtitle/etc. But we've already discussed about those options having different interfaces could be beneficial, so nevermind.